src/Language/Source.php

```diff
@@ -28,7 +33,8 @@ public function __construct($body, $name = null)
         );
 
         $this->body = $body;
-        $this->length = mb_strlen($body, 'UTF-8');
+        $this->buf =  array_map('GraphQL\Utils::ord', preg_split('//u', $body, -1, PREG_SPLIT_NO_EMPTY));
```

Comment:
Yeah, I was suspecting that memory problems are possible with this approach (had previous experience with `preg` functions behaving unpredictably on large strings), but didn't have a chance yet to test it myself. Good that you found this issue before merging.

I guess we should write own utf8 stream to actually resolve the issue.